File: ts/2025/day01.ts
let p1 = 0, p2 = 0
let pos = 50
for await (const line of console) {
	if (line === "") { continue }
	const d = line[0] == "L" ? -1 : 1
	for (let i = 0; i < Number(line.slice(1)); i++) {
		pos = (pos + d) % 100
		if (pos === 0) {
			p2++
		}
	}
	if (pos === 0) {
		p1++
	}
}
console.log("Part 1: %d", p1)
console.log("Part 2: %d", p2)

File: py/2025/day01.py
import sys

p1, p2 = 0, 0
pos = 50
D = sys.stdin.read()
lines = [line for line in D.split("\n") if line]
for line in lines:
    d = -1 if line[0] == "L" else 1
    for i in range(int(line[1:])):
        pos = (pos + d) % 100
        if pos == 0:
            p2 += 1
    if pos == 0:
        p1 += 1

print(f"Part 1: {p1}")
print(f"Part 2: {p2}")
